patches/api/0323-Add-TameableDeathMessageEvent.patch

```diff
@@ -0,0 +1,81 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Owen1212055 <23108066+Owen1212055@users.noreply.github.com>
+Date: Fri, 19 Mar 2021 23:25:38 -0400
+Subject: [PATCH] Add TameableDeathMessageEvent
+
+
+diff --git a/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java b/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java
+new file mode 100644
+index 0000000000000000000000000000000000000000..70cd37aefbd5d64c798ab2fc3b6d502134690348
+--- /dev/null
++++ b/src/main/java/io/papermc/paper/event/entity/TameableDeathMessageEvent.java
+@@ -0,0 +1,69 @@
++package io.papermc.paper.event.entity;
++
++import net.kyori.adventure.text.Component;
++import org.bukkit.entity.Tameable;
++import org.bukkit.event.Cancellable;
++import org.bukkit.event.HandlerList;
++import org.bukkit.event.entity.EntityEvent;
++import org.jetbrains.annotations.NotNull;
++
++/**
++ * Called when a {@link Tameable} dies and sends a death message.
++ */
++public class TameableDeathMessageEvent extends EntityEvent implements Cancellable {
++
++    private static final HandlerList handlers = new HandlerList();
++    private boolean cancelled;
++    private Component deathMessage;
++
++    public TameableDeathMessageEvent(@NotNull Tameable what, @NotNull Component deathMessage) {
++        super(what);
++        this.deathMessage = deathMessage;
++    }
++
++    /**
++     * Set the death message that appears to the owner of the tameable.
++     *
++     * @param deathMessage Death message to appear
++     */
++    public void deathMessage(@NotNull Component deathMessage) {
++        this.deathMessage = deathMessage;
++    }
++
++    /**
++     * Get the death message that appears to the owner of the tameable.
++     *
++     * @return Death message to appear
++     */
++    @NotNull
++    public Component deathMessage() {
++        return deathMessage;
++    }
++
++    @Override
++    public boolean isCancelled() {
```

Comment:
The cancel is for die or only for the message to owner?

Reply:
For the message.

Reply:
Okay, maybe can this be clarify in the docs? or is only me with my confuse xd

Reply:
I believe it's self explanatory that cancelling a Death Message Event will cancel the sending of the message. It wouldn't hurt to explicitly say it at the end of the docs for the class though I suppose.

Reply:
I'm not sure this is needed. Most even cancel methods are not documented.
![image](https://user-images.githubusercontent.com/23108066/127873164-7d6c2d4a-40f6-414e-98c6-74dd9312c5bd.png)
If you look at the top, you will know that the event represents this. And when it is cancelled it will stop this from happening.